catchup/service.go

```diff
@@ -600,6 +600,15 @@ func (s *Service) syncCert(cert *PendingUnmatchedCertificate) {
 
 // TODO this doesn't actually use the digest from cert!
 func (s *Service) fetchRound(cert agreement.Certificate, verifier *agreement.AsyncVoteVerifier) {
+	// is there any point attempting to retrieve the block ?
+	if s.nextRoundIsNotSupported(cert.Round) {
+		// we might get here if the agreement service was seeing the certs votes for the next
+		// block, without seeing the actual block. Since it hasn't seen the block, it couldn't
+		// tell that it's an unsupported protocol, and would try to request it from the catchup.
+		s.handleUnsupportedRound(cert.Round)
```

Comment:
Is it possible to get a syncCert request on an unsupported round when the catchup has not downloaded the last supported block yet?

Reply:
as discussed, the `handleUnsupportedRound` already test that and makes the correct call.